### PR TITLE
BUGFIX: BB-2092 Show custom LoadingComponent in PivotTable

### PR DESCRIPTION
**Pull request by scavnickyj**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7255/ : SUCCESS in 5m 04s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3248/ : SUCCESS in 3s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3249/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3170/ : FAILURE in 2m 15s

**Comment by scavnickyj** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1585.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3250/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3171/ : SUCCESS in 17m 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7256/ : SUCCESS in 5m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3251/ : SUCCESS in 4s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3806/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3255/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3173/ : SUCCESS in 13m 05s

**Comment by scavnickyj** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7260/ : SUCCESS in 5m 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3258/ : SUCCESS in 3s

File: src/components/core/tests/PivotTable.spec.tsx [hunk 1]
```diff
@@ -69,6 +69,12 @@ describe("PivotTable", () => {
         expect(wrapper.find(PivotTableInner)).toHaveLength(1);
     });
 
+    it("should call passed loading component", () => {
+        const CustomLoadingComponent = jest.fn().mockReturnValue(<div />);
+        renderComponent({ LoadingComponent: CustomLoadingComponent });
+        expect(CustomLoadingComponent).toHaveBeenCalled();
+    });
+
```
Comment (xMort) [inline src/components/core/tests/PivotTable.spec.tsx:77]: There should also be an opposite test that test usage of internal LoadingComponent when prop is undefined/null (the default props). I don't see it in the file. The original author probably forgot about it. Can you add it, please?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7266/ : SUCCESS in 5m 05s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3262/ : SUCCESS in 3s

File: src/components/core/tests/PivotTable.spec.tsx (hunk 1 above)
Comment (scavnickyj) [reply src/components/core/tests/PivotTable.spec.tsx:83]: I also rewrote the former one to correspond with the latter.

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3809/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3263/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3176/ : SUCCESS in 14m 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7273/ : SUCCESS in 6m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3276/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3817/ : SUCCESS in 20m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/5/ : SUCCESS in 18s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7276/ : SUCCESS in 7m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1780/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2613/ : SUCCESS in 16m 06s
